cpp/src/io/utilities/trie.cuh

```diff
@@ -83,8 +83,8 @@ trie create_serialized_trie(const std::vector<std::string>& keys, rmm::cuda_stre
 __host__ __device__ inline bool serialized_trie_contains(device_span<serial_trie_node const> trie,
                                                          device_span<char const> key)
 {
-  if (trie.data() == nullptr || trie.empty()) return false;
-  if (key.empty()) return trie.front().is_leaf;
+  if (trie.empty()) { return false; }
```

Comment:
What's wrong with `trie.data() == nullptr`?

Reply:
I assume it's strictly redundant since a device span with `data() == nullptr` will always have size 0.